src/Ledger/Conway/Specification/Epoch.lagda

```diff
@@ -446,6 +447,21 @@ its results by carrying out each of the following tasks.
 
 \begin{NoConway}
 \begin{figure*}[ht]
+\begin{code}[hide]
+calculatePoolDistr : Snapshot → PoolDistr
+calculatePoolDistr ss =
+    zipWithᵐ (λ c pp → (? , ?) ) sd (poolParameters ss)
+  where
+    open Snapshot
+    totalStake = ∑[ c ← stake ss ] c
+
+    sd : KeyHash ⇀ Coin
+    sd = aggregateBy (delegations ss ˢ) (stake ss)
+
+    zipWithᵐ : {A B C D : Type} → (B → C → D) → A ⇀ B → A ⇀ C → A ⇀ D
+    zipWithᵐ = ?
```

Comment:
Perhaps I need to implement this from scratch (?)

Reply:
One idea to implement it:
```
map fromJust $ filter isJust $ unionWith (fold (const nothing) (const nothing) (just . uncurry _,_))
```
Some fiddling will be necessary to justify the `fromJust` call.

Reply:
Went with the later one.